### PR TITLE
Fix typos in Docs/Numerics/Poisson solvers

### PR DESCRIPTION
**Pull request by navidcy**

### CODE REVIEW

**Review by glwagner — APPROVED** [review]
<3 <3 <3 <3

File: docs/src/numerical_implementation/poisson_solvers.md [hunk 1]
```diff
@@ -53,21 +53,21 @@ The eigenvalues are given by [Schumann88](@cite) and can also be tediously deriv
 discrete Fourier transform into \eqref{eq:poisson-spectral}:
 ```math
 \begin{align}
-    \lambda^x_i &= 4\frac{N_x^2}{L_x^2} \sin^2 \left [ \frac{(i-1)\pi}{N_x}  \right ], \quad i=0,1, \dots,N_x-1 \, , \\
-    \lambda^x_j &= 4\frac{N_y^2}{L_y^2} \sin^2 \left [ \frac{(j-1)\pi}{N_y}  \right ], \quad j=0,1, \dots,N_y-1 \, , \\
-    \lambda^x_k &= 4\frac{N_z^2}{L_z^2} \sin^2 \left [ \frac{(k-1)\pi}{2N_z} \right ], \quad k=0,1, \dots,N_z-1 \, ,
+    \lambda^x_i &= 4\frac{N_x^2}{L_x^2} \sin^2 \left [ \frac{(i-1) \pi}{N_x}  \right ], \quad i=0, 1, \dots, N_x-1 \, , \\
+    \lambda^y_j &= 4\frac{N_y^2}{L_y^2} \sin^2 \left [ \frac{(j-1) \pi}{N_y}  \right ], \quad j=0, 1, \dots, N_y-1 \, , \\
+    \lambda^z_k &= 4\frac{N_z^2}{L_z^2} \sin^2 \left [ \frac{(k-1) \pi}{2N_z} \right ], \quad k=0, 1, \dots, N_z-1 \, ,
```
Comment (navidcy) [inline docs/src/numerical_implementation/poisson_solvers.md:58]: is the `2 N_z` in the denominator here a typo?

Comment (glwagner) [reply docs/src/numerical_implementation/poisson_solvers.md:58]: Oy, I don't know. Are these eigenvalues only for Periodic, Periodic, Bounded?

Comment (glwagner) [reply docs/src/numerical_implementation/poisson_solvers.md:58]: Periodic is

https://github.com/CliMA/Oceananigans.jl/blob/56fe1aa046e8f9d485c00719de523af01ffcb495/src/Solvers/poisson_eigenvalues.jl#L10

Bounded is

https://github.com/CliMA/Oceananigans.jl/blob/56fe1aa046e8f9d485c00719de523af01ffcb495/src/Solvers/poisson_eigenvalues.jl#L22

**Pull request merged**